crates/test-programs/wasi-tests/src/bin/path_link_nonstrict.rs

```diff
@@ -0,0 +1,278 @@
+    );
+    wasi::path_unlink_file(dir_fd, "symlink").expect("removing a symlink");*/
+
+    // Create a link to a symlink loop
+    // NON-STRICTNESS
+    // Windows doesn't support symlink loops
+    /*wasi::path_symlink("symlink", dir_fd, "symlink").expect("creating a symlink loop");
+
+    assert_eq!(
+        wasi::path_link(dir_fd, 0, "symlink", dir_fd, "link")
+            .expect_err("creating a link to a symlink loop")
+            .raw_error(),
+        wasi::ERRNO_LOOP,
+        "errno should be ERRNO_LOOP"
+    );
+    wasi::path_unlink_file(dir_fd, "symlink").expect("removing a symlink");
+
+    // Create a link where target is a dangling symlink
+    wasi::path_symlink("target", dir_fd, "symlink").expect("creating a dangling symlink");
+
+    assert_eq!(
+        wasi::path_link(dir_fd, 0, "file", dir_fd, "symlink")
+            .expect_err("creating a link where target is a dangling symlink")
+            .raw_error(),
+        wasi::ERRNO_EXIST,
+        "errno should be ERRNO_EXIST"
+    );
+    wasi::path_unlink_file(dir_fd, "symlink").expect("removing a symlink");
+
+    // Create a link to a file following symlinks
+    wasi::path_symlink("file", dir_fd, "symlink").expect("creating a valid symlink");
+    wasi::path_link(
+        dir_fd,
+        wasi::LOOKUPFLAGS_SYMLINK_FOLLOW,
+        "symlink",
+        dir_fd,
+        "link",
+    )
+    .expect("creating a link to a file following symlinks");
+    link_fd = open_link(dir_fd, "link");
+    check_rights(file_fd, link_fd);
+    wasi::path_unlink_file(dir_fd, "link").expect("removing a link");
+    wasi::path_unlink_file(dir_fd, "symlink").expect("removing a symlink");
+
+    // Create a link where target is a dangling symlink following symlinks
+    wasi::path_symlink("target", dir_fd, "symlink").expect("creating a dangling symlink");
+
+    assert_eq!(
+        wasi::path_link(
+            dir_fd,
+            wasi::LOOKUPFLAGS_SYMLINK_FOLLOW,
+            "symlink",
+            dir_fd,
+            "link",
+        )
+        .expect_err("creating a link where target is a dangling symlink following symlinks")
+        .raw_error(),
+        wasi::ERRNO_NOENT,
+        "errno should be ERRNO_NOENT"
+    );
+    wasi::path_unlink_file(dir_fd, "symlink").expect("removing a symlink");
+
+    // Create a link to a symlink loop following symlinks
+    wasi::path_symlink("symlink", dir_fd, "symlink").expect("creating a symlink loop");
+
+    assert_eq!(
+        wasi::path_link(
+            dir_fd,
+            wasi::LOOKUPFLAGS_SYMLINK_FOLLOW,
+            "symlink",
+            dir_fd,
+            "link",
+        )
+        .expect_err("creating a link to a symlink loop following symlinks")
+        .raw_error(),
+        wasi::ERRNO_LOOP,
+        "errno should be ERRNO_LOOP"
+    );
+    wasi::path_unlink_file(dir_fd, "symlink").expect("removing a symlink");
+
+    // Clean up.
+    wasi::path_unlink_file(dir_fd, "file").expect("removing a file");
+    */
```

Comment:
Hmm, I think I'd prefer if we had two versions of the testcase, not necessarily overlapping in any way, but such that the "strict" tests are executed only on supported *nix platform, while the rest always executes the "nonstrict" set.

Reply:
Done, the non-strict version is no more!